### ISSUE TITLE
Compilation to WASM runtime

### ISSUE THREAD

**Issue by vertexclique** [bastion-rs/bastion#134]
Make everything ready for WASM runtime. The executor should have single thread. This is requested by various people to incorporate Bastion to their projects.

**Comment by zmanian** [comment]
WASM should enable bastion-rs to offer something very similar to Erlang in terms of being able to distribute both code and data to distributed processes.

Erlang was designed to make functions serializable over the network allowing the persona of particular worker to evolve over time. 

WASM allow you essentially do the same thing.

It seems inevitable to me that there will be WASM based Hadoop/Spark/Erlang replacement and Bastion seems like it's moving in that direction.

**Comment by o0Ignition0o** [comment]
Ok I have no idea how to do it, but I wanna give it a try.

**Comment by Stupremee** [comment]
I'm working on a wasm executor in agnostik that we will need to add wasm support to bastion. It will be in the next agnostik release

**Comment by vertexclique** [comment]
It is not possible until `bastion` package can be compiled solely for wasm target. Placing executor doesn't mean that it will compile to wasm.

**Comment by zmanian** [comment]
Yeah, I think the idea was executing WASM functions via the bastion runtime.

**Comment by attliaLin** [comment]
Nice idea. Maybe provide the ability for the other script languages like Lua, python as the executing runtime.
